server/sonar-server/src/main/java/org/sonar/server/computation/batch/BatchReportReader.java

```diff
@@ -27,6 +28,9 @@
 public interface BatchReportReader {
   BatchReport.Metadata readMetadata();
 
+  @CheckForNull
+  Reader readScannerLogs();
```

Comment:
I would also prefer an method where reading the file is kept the responsibility of the BatchReportReader.
In the end, you are consuming the reader to read data line by line, this method should return a `ClosableIterator` of `String`
In addition, it will be much easier to use in unit tests